core/object/object.h

```diff
@@ -80,6 +80,7 @@ enum PropertyHint {
 	PROPERTY_HINT_INT_IS_OBJECTID,
 	PROPERTY_HINT_INT_IS_POINTER,
 	PROPERTY_HINT_ARRAY_TYPE,
+	PROPERTY_HINT_DICTIONARY_TYPE,
```

Comment:
I think it's better to add the hint to the end of the enum, without shifting the values.